rust/yarp-sys/src/lib.rs

```diff
@@ -0,0 +1,36 @@
+#![allow(non_upper_case_globals)]
+#![allow(non_camel_case_types)]
+#![allow(non_snake_case)]
+
+use std::ffi::c_char;
+
+// In `build.rs`, we use `bindgen` to generate bindings based on C headers and `librubyparser`. Here
+// is where we pull in those bindings and make them part of our library.
+include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
```

Comment:
typically when using `bindgen`, I stick the `include!` in a module so we can scope the `allow` pragmas to only the generated code. Something like this:

```rust
#[allow(non_upper_case_globals)]
#[allow(non_camel_case_types)]
#[allow(non_snake_case)]
mod bindings {
    // In `build.rs`, we use `bindgen` to generate bindings based on C headers and `librubyparser`. Here
    // is where we pull in those bindings and make them part of our library.
    include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
}

pub use self::bindings::*;
```

Reply:
I don't really feel that strongly about this either way, although I'd kinda lean towards leaving it as-is since ideally, IMO, this whole crate would be bindgen code. I only have the two manually defined types in there (`yp_string_t` and `yp_string_t_type`) because I don't know how to get bindgen to generate an embedded enum as something other than the via its default naming (`yp_string_t__bindgen_ty_1` in this case) and that doesn't seem ideal to me.

It's been over 6 weeks since I've worked on any of this code, so I'm forgetting what I came across, but IIRC, I came across something that led me to believe that it was (or would be?) doable. If we end up with needing to define type manually for the long term, then yeah, that makes sense to me; otherwise we end up with a `bindings` submodule with no peer modules and that seems unnecessary.

Reply:
this forces allowing rustc "warn by default" lints for the entire crate, including any unit tests, should they be added in the future.

Reply:
I don't think I see the problem with that, but at this point I'm in favor of getting this merged since I don't have much time to work on this right now (I did 6 weeks ago, but life has changed a bit since then), so I'll just change it.